examples/cutlass/kaasComplex/test.py

```diff
@@ -0,0 +1,163 @@
+
+    # The kernel takes a Params struct as an argument (by value). Rather than
+    # try to define that struct in python, we instead find its size manually
+    # (in cuda using sizeof()) and then specify a byte array of the same size
+    # here. Pycuda doesn't care about the type in practice, it only needs the
+    # size. This type string is defined by python's "struct" module.
+    cutlassKern.prepare("328s")
+
+    refKern = mod.get_function("ReferenceGemm_kernel")
+    # See python's struct module for a description of this type string
+    refKern.prepare("iiifPiPifPi")
+
+    return refKern, cutlassKern
+
+
+class kernelConfig(ct.Structure):
+    """This mirrors the CudaConfig struct defined in cutlassAdapters.h"""
+    _fields_ = [
+        ("gridX", ct.c_int),
+        ("gridY", ct.c_int),
+        ("gridZ", ct.c_int),
+        ("blockX", ct.c_int),
+        ("blockY", ct.c_int),
+        ("blockZ", ct.c_int),
+        ("smem_size", ct.c_int)
+    ]
+
+
+def loadAdapter():
+    libc = ct.cdll.LoadLibrary("./cutlassAdapters.so")
+    getArg = libc.adaptSGEMMArgs
+    getArg.argtypes = [ct.c_int, ct.c_int, ct.c_int, ct.c_float, c_complex_p, ct.c_int,
+                       c_complex_p, ct.c_int, ct.c_float, c_complex_p, ct.c_int]
+    # getArg.argtypes = [ct.c_int, ct.c_int, ct.c_int, ct.c_float, ct.c_void_p, ct.c_int, 
+    #                     ct.c_void_p, ct.c_int, ct.c_float, ct.c_void_p, ct.c_int]
+    # Instead of trying to define the Params struct in python, we just pretend
+    # that it's a byte array of the same size (320 bytes in this case)
+    getArg.restype = ct.POINTER(ct.c_byte*328)
+
+    getDims = libc.getCudaConfig
+    # M, N, K
+    getDims.argtypes = [ct.c_int, ct.c_int, ct.c_int]
+    getDims.restype = ct.POINTER(kernelConfig)
+
+    return (getArg, getDims)
+
+
+def testKern():
+    """This kernel is useful for trying out ideas without all the complexity of
+    cutlass"""
+    mod = cuda.module_from_file("./cutlass.cubin")
+    kern = mod.get_function("testKernel")
+    kern.prepare("16s")
+
+    lib = ct.cdll.LoadLibrary("./cutlassAdapters.so")
+    getStruct = lib.getTestStruct
+    getStruct.argtypes = [ct.c_int, ct.POINTER(complex)]
+    getStruct.restype = ct.POINTER(ct.c_byte*16)
+
+    anInt = 42
+    hArr = np.arange(4096*2, dtype=np.csingle) * (1+1j)
+    print("numpy sum is")
+    print(np.sum(hArr))
+
+    dArr = cuda.mem_alloc(hArr.nbytes)
+    cuda.memcpy_htod(dArr, hArr)
+
+    arg = getStruct(anInt, ct.cast(int(dArr), ct.POINTER(complex)))
+
+    kern.prepared_call((1, 1, 1), (1, 1, 1), arg.contents)
+
+
+def testSgemm(M, N, K, alpha, beta):
+    lda = M
+    ldb = K
+    ldc = M
+
+    getArg, getDims = loadAdapter()
+    refKern, cutlassKern = loadKerns()
+
+    rng = np.random.default_rng(5)
+    a = np.asfortranarray(rng.random((M, K), dtype=np.float32) * (1+1j))
```

Comment:
not all that important I guess, but we may as well make completely random complex numbers. What's here will have a random real part with a non-random imaginary part. Try this instead:
np.asfortranarray(rng.random((M, K), dtype=np.float32) * (rng.random((M,K), dtype=np.float32)*1j))